crates/bitwarden/src/tool/generators/username_forwarders/firefox.rs

```diff
@@ -0,0 +1,54 @@
+use reqwest::{
+    header::{self},
+    StatusCode,
+};
+
+use crate::error::{Error, Result};
+pub async fn generate(
+    http: &reqwest::Client,
+    api_token: String,
+    website: Option<String>,
+) -> Result<String> {
+    if api_token.is_empty() {
+        return Err(Error::Internal("Invalid Firefox Relay API API key"));
+    }
+
+    #[derive(serde::Serialize)]
+    struct Request {
+        enabled: bool,
+        generated_for: Option<String>,
+        description: String,
+    }
+
+    let description = website
+        .as_ref()
+        .map(|w| format!("{w} - "))
+        .unwrap_or_default();
+    let description = format!("{description}Generated by Bitwarden.");
```

Comment:
We seem to be duplicating this in many of the handlers. Should we pre-process this as part of the input?

Reply:
I've extracted it to a function at least, as not all the services use it. I've noticed that at the moment Firefox is using a slightly different message. I don't know if that matters, but I defined a separate function for it.